File: typescript/src/services/export-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { ExportPageDto, ExportSpaceDto } from '../models/index.js';


export class ExportApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async exportPage(request: ExportPageDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/export`, request);
  }

  async exportSpace(request: ExportSpaceDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`spaces/export`, request);
  }

}

File: c#/src/Docmost.Sdk/Services/ExportApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class ExportApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> ExportPageAsync(ExportPageDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/export", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> ExportSpaceAsync(ExportSpaceDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("spaces/export", request, cancellationToken).ConfigureAwait(false);
    }

}
